src/AppInstallerCLIE2ETests/Interop/UpgradeInterop.cs

```diff
@@ -0,0 +1,230 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
```

Comment:
If we're working on tests, this is a good time, but I'll let @AmelBawa-msft decide if it would impact any other pending work.

Reply:
@Trenly, @denelon thank you for the suggestion. I created an internal task to track this: https://task.ms/40439605.
The change will likely be considered in a separate PR since this one focus on implementing a strict subset of the existing test cases (InstallCommand, UninstallCommand and UpgradeCommand) but using the [COM API](https://github.com/microsoft/winget-cli/blob/master/src/Microsoft.Management.Deployment/PackageManager.idl).